### PR TITLE
Add WCMP test plan for community

### PR DESCRIPTION
**Pull request by weiguo-nvidia**
Add WCMP test plan

### CODE REVIEW

File: doc/wcmp/WCMP-test-plan.md [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+
+### Show
+The following command can be used to show WCMP status:
+```
+show bgp device-global
+show bgp device-global --json
+```
+Example:
+```
+admin@r-ocelot-07:~$ show bgp device-global
```
Comment (nhe-NV) [inline doc/wcmp/WCMP-test-plan.md]: we could not use the r-ocelot-07:

File: doc/wcmp/WCMP-test-plan.md [hunk 2]
```diff
@@ -0,0 +1,320 @@
+# WCMP Test Plan
+
+## Related documents
+
+| **Document Name** | **Link** |
+|-------------------|----------|
+| SONiC Weighted ECMP | [https://github.com/nazariig/SONiC/blob/773aacaeb7a6e5507af667898957b2060a78280e/doc/wcmp/wcmp-design.md]|
```
Comment (nhe-NV) [inline doc/wcmp/WCMP-test-plan.md]: Do we already have the public one?

File: doc/wcmp/WCMP-test-plan.md [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+```
+Example:
+```
+admin@r-ocelot-07:~$ show bgp device-global
+TSA       WCMP
+--------  -------
+disabled  enabled
+```
+```
+admin@r-ocelot-07:~$ show bgp device-global --json
```
Comment (nhe-NV) [inline doc/wcmp/WCMP-test-plan.md]: r-ocelot-07 issue

File: doc/wcmp/WCMP-test-plan.md [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+
+
+## Test structure
+DUT advertises BGP route prefix to SPINE1 (red line) and SPINE2 (blue line), then SPINE1 and SPINE2 forward the route to Remote host. So there are 4 nexthops to the DUT on the remote host routing table.
+
+![Use case scenario](images/WCMP_topology.png)
+
+- If WCMP is `disabled`, the weight value of route on remote host is 1
+```
+admin@mtvr-moose-02:~$ ip route show 100.0.0.1
```
Comment (nhe-NV) [inline doc/wcmp/WCMP-test-plan.md]: mtvr-moose-02: can not be shown

File: doc/wcmp/WCMP-test-plan.md (hunk 2 above)
Comment (weiguo-nvidia) [reply doc/wcmp/WCMP-test-plan.md]: I have deleted this part

File: doc/wcmp/WCMP-test-plan.md [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+### Test case #1 - WCMP command line test
+1. Enable WCMP and verify WCMP status
+```
+admin@router:~$ sudo config bgp device-global wcmp enabled
+admin@router:~$ show bgp device-global
+TSA       WCMP
+--------  -------
+disabled  enabled
+```
+2. Verify BGP link bandwidth extended community is added in FRR
```
Comment (nazariig) [inline doc/wcmp/WCMP-test-plan.md]: @weiguo-nvidia we do have it in a UTs

File: doc/wcmp/WCMP-test-plan.md [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+```
+4. Disable WCMP and verify WCMP status
+```
+admin@router:~$ sudo config bgp device-global wcmp disabled
+admin@router:~$ show bgp device-global
+TSA       WCMP
+--------  -------
+disabled  disabled
+```
+5. Verify BGP link bandwidth extended community is deleted in FRR
```
Comment (nazariig) [inline doc/wcmp/WCMP-test-plan.md]: @weiguo-nvidia we do have it in a UTs

File: doc/wcmp/WCMP-test-plan.md [hunk 7]
```diff
@@ -0,0 +1,295 @@
...
+6. Verify the route weight value on remote host is 1, which means WCMP is disabled
+```
+admin@router:~$ ip route show 100.0.0.1
+100.0.0.1 proto bgp metric 20
+        nexthop via 10.10.26.1 dev Ethernet100 weight 1
+        nexthop via 10.10.27.1 dev Ethernet104 weight 1
+        nexthop via 10.10.28.1 dev Ethernet108 weight 1
+        nexthop via 10.10.29.1 dev Ethernet112 weight 1
+```
+7. Config WCMP with invalid parameter, verify config fail and report error message
```
Comment (nazariig) [inline doc/wcmp/WCMP-test-plan.md]: @weiguo-nvidia we do have it in a UTs

File: doc/wcmp/WCMP-test-plan.md [hunk 8]
```diff
@@ -0,0 +1,295 @@
...
+7. Recover the failed link between DUT and SPINE1
+8. Verify the route weight value on remote host is 25
+
+### Test case #5 - Verify configuration persists after warm/cold/fast reboot
+1. Enable WCMP on DUT, verify the config take effective
+2. DUT warm/cold/fast reboot
+3. Verify WCMP configuration persist after reboot
+4. Traffic test, verify the WCMP function still take effect
+
+### Test case #6 - Verify configuration persists after warm/cold/fast reboot
```
Comment (nazariig) [inline doc/wcmp/WCMP-test-plan.md]: @weiguo-nvidia the same topic twice. A typo?

File: doc/wcmp/WCMP-test-plan.md [hunk 9]
```diff
@@ -0,0 +1,295 @@
...
+7. Config WCMP with invalid parameter, verify config fail and report error message
+```
+admin@router:~$ sudo config bgp device-global wcmp 1
+Usage: config bgp device-global wcmp [OPTIONS] COMMAND [ARGS]...
+Try "config bgp device-global wcmp -h" for help.
+
+Error: No such command "1".
+```
+
+### Test case #2 - Config WCMP from config_db.json file
```
Comment (nazariig) [inline doc/wcmp/WCMP-test-plan.md]: @weiguo-nvidia no need to repeat the same twice: this test case only makes sense to catch an error message produced by a daemon when invalid configuration is provided

**Review by nazariig — CHANGES_REQUESTED** [review]

File: doc/wcmp/WCMP-test-plan.md (hunk 6 above)
Comment (weiguo-nvidia) [reply doc/wcmp/WCMP-test-plan.md]: Remove this step

File: doc/wcmp/WCMP-test-plan.md (hunk 7 above)
Comment (weiguo-nvidia) [reply doc/wcmp/WCMP-test-plan.md]: Remove this step

File: doc/wcmp/WCMP-test-plan.md (hunk 8 above)
Comment (weiguo-nvidia) [reply doc/wcmp/WCMP-test-plan.md]: Update the case name

File: doc/wcmp/WCMP-test-plan.md (hunk 9 above)
Comment (weiguo-nvidia) [reply doc/wcmp/WCMP-test-plan.md]: Remove this case

File: doc/W-ECMP/W-ECMP-test-plan.md [hunk 10]
```diff
@@ -0,0 +1,196 @@
...
+### Test case #1 - W-ECMP command line test
+1. Enable W-ECMP and verify W-ECMP status
+```
+admin@router:~$ sudo config bgp device-global wcmp enabled
+admin@router:~$ show bgp device-global
+TSA       WCMP
+--------  -------
+disabled  enabled
+```
+2. Verify the route weight value on remote host is 25, which means W-ECMP is effect
```
Comment (nazariig) [inline doc/W-ECMP/W-ECMP-test-plan.md:105]: @weiguo-nvidia FYI. Need to update the test plan, once the FRR normalization to 255 is backported

**Review by nazariig — APPROVED** [review]

**Pull request closed without merging**